crates/namada/src/ledger/native_vp/masp.rs

```diff
@@ -304,110 +296,327 @@ where
             }
         }
 
-        // Verify the changes to balance keys and return the transparent
-        // transfer data Get the token from the balance key of the MASP
-        let balance_addresses: Vec<[&Address; 2]> = keys_changed
+        let mut result = ChangedBalances::default();
+        // Get the changed balance keys
+        let counterparts_balances: Vec<_> = keys_changed
             .iter()
             .filter_map(is_any_shielded_action_balance_key)
             .collect();
```

Comment:
shouldn't need to collect I think
```suggestion
        // Get the changed balance keys
        let counterparts_balances = keys_changed
            .iter()
            .filter_map(is_any_shielded_action_balance_key);
```

Reply:
Thanks, good catch. Addressed this in #3264 .